Bump Eclipse to e4.30 and Tycho to 4.0.4

cmark opened the pull request:
This PR bumps the target platform to Eclipse Platform 4.30 and Tycho to 4.0.4.

The following dependency changes have been applied:
* Upgrade Elasticsearch to 8.12.1
* Upgrade Xtext to 2.33.0
* Upgrade EMF to 2.35.0
* Upgrade MWE2 to 2.16.0
* Upgrade ECL to 2.1.5
* Upgrade SLF4J to 2.0.11
* Upgrade Logback to 1.4.14
* Upgrade Jackson to 2.16.1
* Upgrade Jetty to 12.0.3
* Upgrade to Jakarta EE10
* Upgrade Spring to 6.1.3
* Upgrade springdoc to 2.3.0
* Upgrade swagger to 2.2.20
* Upgrade Guava to 32.1.3-jre
* Upgrade OWLAPI to 4.5.26
* Upgrade Protege to custom 5.0.7
* Upgrade snomed-owl-toolkit to a custom 5.1.1
* Upgrade Fastutil to 8.5.12
* Upgrade RestAssured to 4.5.1
* Upgrade Mockito to 5.9.0
* Upgrade ByteBuddy to 1.14.11
* Upgrade uuid-creator to 5.3.7
* Upgrade failsafe to 3.3.2
* Upgrade rxjava to 2.2.21
* Upgrade netty to 4.1.106.Final
* Upgrade kotlin-osgi-bundle to 1.9.22
* Upgrade mapdb to 3.1.0
* Upgrade failsafe to 3.2.2
* Upgrade micrometer to 1.12.2
* Upgrade zjsonpatch to 0.4.16
* Upgrade bucket4j to 8.7.0
* Upgrade hibernate validator to 8.0.1.Final
* Upgrade java-jwt to 4.4.0
* Upgrade jwks-rsa to 0.22.1
* Upgrade testcontainers to 1.19.4
* Add required Apache Commons libraries.

Build changes:
* Add mvn wrapper
* Tycho is now 4.0.4
* jacoco is now 0.8.11
* Remove unused jars from the produced packages

Jira: https://snowowl.atlassian.net/browse/SO-5973

cmark committed:
feat(build): bump tycho to 4.0.4

cmark committed:
refactor: sync all .classpath files with latest eclipse/m2e changes

cmark committed:
ci: bump gh action packages to their latest version

cmark committed:
build: add mvn wrapper with version 3.9.5

cmark committed:
docs: update Building from source and Development sections

cmark committed:
build: use maven wrapper when building on GitHub

cmark committed:
refactor: sync test.dependencies/.classpath

cmark committed:
fix: start migrating to newer, 5.0 snomed-owl-toolkit version

cmark committed:
build: remove snomed-owl-toolkit 3.0.6 from tp

cmark committed:
chore(deps): depend on Guava 32.1.3-jre

Drop javax.inject libraries in favor of jakarta.inject (Guice 7.0).

cmark committed:
chore(deps): depend on Spring 6.1.3

Migrate to springdoc openapi 2.3.0

cmark committed:
refactor(rest): resolve compile errors after moving to SpringDoc 2.3.0

cmark committed:
fix(index): resolve test failures in index tests

cmark committed:
fix(api): migrate SnowOwlSecurityConfig to component based...

...security configuration

cmark committed:
fix(api): use the new OpenApiService class to access the OpenAPI bean

cmark committed:
fix(dev): core-e2e-tests.launch configuration plugin list

cmark committed:
test: add jetty-ee8 variant of core-e2e-tests launch config

cmark committed:
fix(test): make sure we try to deploy the rest api into the ee10...

...jetty environment

Fix jetty 12 osgi boot configuration issue by copying files and fixing
the erroneous section.

cmark committed:
fix(test): use PathUtils instead of deprecated FileUtils

cmark committed:
Merge branch '9.x' into bump/eclipse-tycho-build

cmark committed:
fix(core): remove unused PlatformUtil.enableSystemProxies()

Remove dependency on eclipse.core.net.

cmark committed:
test: add a trace log entry after a bundle successfully started

cmark committed:
test(dev): fix core-e2e-tests launch config on Jetty 12 EE10

cmark committed:
fix(snomed): remove unnecessary eclipse.core.databinding dependency...

...from snomed core module

cmark committed:
fix(log): fix reference to moved jetty annotation parser class

cmark committed:
fix(api): replace Spring versions in MANIFEST.MF

cmark committed:
fix(api): minor changes in core.rest pom.xml

cmark committed:
fix(api): use newer xmlns in web.xml

cmark committed:
fix(pom): remove unused tycho extras version prop

cmark committed:
fix: minor api module fixes

cmark committed:
fix(dev): remove m2e classpath container from all bundles

cmark committed:
fix(api): add swagger-core-jakarta 2.2.20 to core.rest dependencies

cmark committed:
feat(deps): bump Jackson dependency constraints to [2.16.1,3.0.0)

cmark committed:
feat(deps): bump slf4j to 2.0.0 and logback to 1.4.14

cmark committed:
feat(api): dependencies in core.rest module to allow Spring 6 to boot

cmark committed:
fix(log): disable status messages from logback when booting

Removed unknown jmxConfigurator entry.

cmark committed:
fix(api): make sure we rely only on the declared web.xml...

...when initializing the core.rest module

This prevents Jetty to use its internal slow class scanning ability to
detect various servlet api implementations.

cmark committed:
fix(api): remove deprecated timestamp parameter from...

...get res.collection api

cmark committed:
fix(collections): resolve jackson compile errors after moving to 2.16

cmark committed:
fix(api): enable StandardServletMultipartResolver temporarily

File upload size configurations supported by the now deprecated commons
based file upload resolver must be restored somehow.

cmark committed:
fix(api): add back EnableWebSecurity annotation

cmark committed:
fix(test): make sure we start the test osgi environment in the correct..

...order

First, ensure that slf4j and logback along with spifly is auto-started
in OSGi level 2 otherwise logging won't work.
Second, ensure ee10.webapp is started automatically so that spifly
detects and registers ServiceLoader mechanism for the webapp
Configurations class, which is responsible for handling various web
application configuration types, eg. web.xml.
Third, ensure that osgi.boot is also started before we start the webapp
otherwise the system can randomly throw an error about unregistered
bundleresource filesystem provider.

Then boot Jetty in OSGi mode and deploy our rest api module.

cmark committed:
fix(test): resolve ambiguous uri errors in core e2e tests

cmark committed:
test: fix index launch configurations

Bump ES 8 images to 8.12.0

cmark committed:
test: fix commons launch configuration

cmark committed:
test: fix core unit test launch config

Remove no longer needed EE8 container based core e2e launch config.

cmark committed:
test: fix eventbus launch config

cmark committed:
test: fix issues in launch configs as much as possible

Remaining problems:
* owlapi/protege guava import issue
* test.commons dependency often requires full API infrastructure to be
present even for low-level unit tests

cmark committed:
chore(deps): fix httpclient require-bundle version ranges

cmark committed:
fix(dev): resolve issues in validation and reasoner unit test configs

cmark committed:
fix(deps): move to snomed-owl-toolkit 5.1.0

Use only the necessary transitive dependencies that are required to the
axiom converter service and nothing else to minimize future security
impacts.

cmark committed:
fix(core): ensure that in non-snowowl environments our command provider.

...won't crash the entire console help output

cmark committed:
chore(deps): make sure spring-expression lib is included in web-inf/lib

cmark committed:
chore(deps): custom hamcrest wrapper is no longer needed...

...to server 2.x hamcrest as 1.x

cmark committed:
fix(snomed): resolve owltoolkit errors by moving to our customized...

...version

cmark committed:
feat(test): introduce bootRestApi helper to ensure consistent...

...boot order of jetty and rest api bundles in e2e tests

cmark committed:
test(snomed): resolve boot issues in snomed-e2e-tests

cmark committed:
fix(api): missing `force` parameter annotation value reported by Spring

cmark committed:
chore(core): print actual syntax error when failing to optimize clause

cmark committed:
test: disable automatic url encoding in rest-assured

cmark committed:
feat(core): migrate to bucket4j-core:8.7.0

cmark committed:
test(core): resolve test failures due to disable url encoding

Add helper `RestExtensions.encodeQueryParameters(Map)` to aid URL
encoding in test scenarios where needed.

cmark committed:
feat(core): migrate to zjsonpatch:0.4.16

Bump commons-collections4 to 4.4

cmark committed:
feat(core): migrate to jgrapht-core:1.5.2

cmark committed:
feat(core): migrate to micrometer:1.12.2

cmark committed:
feat(core): migrate to jakarta validation and hibernate validator 8.0.1

Use jakarta.el and glassfish expressly for expression evaluation.

cmark committed:
chore(core): replace javax.validation import packages with jakarta...

...validation

Replace deprecated hibernate NotEmpty constraint with jakarta
equivalent.

cmark committed:
chore(fhir): replace javax.validation import packages with jakarta...

...validation

Replace deprecated hibernate NotEmpty constraint with jakarta
equivalent.

cmark committed:
chore(snomed): replace javax.validation import packages with jakarta...

...validation

Replace deprecated hibernate NotEmpty constraint with jakarta
equivalent.

cmark committed:
test: fix test failures due to changed bean validation error message

cmark committed:
chore(dev): fix launch configurations after moving to jakarta validation

Resolve test failure in LongOpenHashSetTest.

cmark committed:
chore(dev): fix fhir unit test failures...

...after moving to jakarta validation

cmark committed:
build: fix tycho document plugin version

cmark committed:
test: fix failure in snomed-e2e-tests (they all green now :yey:)

cmark committed:
feat(core): bump java-jwt to 4.4.0 and jwks-rsa to 0.22.1

cmark committed:
fix(commons): add missing comma to commons/pom.xml dep-copy list

cmark committed:
feat(test): bump testcontainers to 1.19.4

Bump system-rules to 1.19.0 and fetch it from maven central.

cmark committed:
test: resolve docker boot issues when running ES8 index tests

cmark committed:
feat(index): migrate failsafe to 3.3.2

cmark committed:
feat(core): migrate failsafe to 3.3.2

cmark committed:
chore(snomed): resolve remaining test launch configuration issues

cmark committed:
fix(fhir): resolve test failures in fhir-api-tests

cmark committed:
feat(commons): replace embedded collections4 jar with target-platform...

...dependency

cmark committed:
feat(core): remove unnecessary xtext.util explicit dependency

cmark committed:
feat(tp): target platform upgrade to Eclipse Platform 4.30

Upgrade Xtext to 2.33.0
Upgrade EMF to 2.35.0
Upgrade MWE2 to 2.16.0
Upgrade ECL to 2.1.5
Upgrade SLF4J to 2.0.11
Upgrade Logback to 1.4.14
Upgrade Jackson to 2.16.1
Upgrade Jetty to 12.0.3
Upgrade to Jakarta EE10
Upgrade OWLAPI to 4.5.26
Upgrade Protege to custom 5.0.7
Upgrade Fastutil to 8.5.12
Upgrade RestAssured to 4.5.1
Upgrade Mockito to 5.9.0
Upgrade ByteBuddy to 1.14.11
Upgrade uuid-creator to 5.3.7
Upgrade failsafe to 3.3.2
Upgrade rxjava to 2.2.21
Add required Apache Commons libraries.

cmark committed:
build: update version restrictions in root pom

cmark committed:
fix(commons): update jheaps version in commons/MANIFEST.MF

cmark committed:
fix(deps): update core.dependencies.feature with updated deps

cmark committed:
chore(ldap): simplify ldap MANIFEST.MF

cmark committed:
chore(dev): sync all launch configurations with latest dependencies

cmark committed:
fix(tp): add back tagsoup library

cmark committed:
fix(cis): resolve feature resolution errors in snomed.cis.feature

cmark committed:
chore(dev): simplify embedded dependency updates by using newer...

...m2e syntax for lifecycle-mappings

cmark committed:
build: configure bundle start levels of logging and jetty to...

...properly boot the server product

Minor feature dependency changes.

cmark committed:
build: resolve test runtime errors when executing them during the build

cmark committed:
build: resolve api test runtime initialization errors

cmark committed:
build: temporarily disable api tests until...

...https://github.com/jetty/jetty.project/issues/11377 gets resolved

cmark committed:
build: add logback/spifly bundle start levels to remaining test configs

cmark committed:
fix(index): use asm 9.6 from target platform instead of embedding...

...into the index bundle

cmark committed:
chore(dev): ignore tycho-consumer-pom.xml files in git dir

cmark committed:
build: fix test.feature bundle list

Custom hamcrest bundle has been replaced with official 2.2.0 version

cmark committed:
chore(commons): hibernate validator has been moved to...

...a different sub group (both groups work for a while, but start using
the new one to avoid build issues in the future)

cmark committed:
build: ignore locally built artifacts when resolving module deps

cmark committed:
build: update jetty xmls to their latest versions

cmark committed:
build: update snowowl startup scripts to use the newer equinox launcher

cmark committed:
chore(api): update caniuse browser compatibility db

cmark committed:
fix(api): resolve npm reported vulnerability issues via `npm audit fix`

cmark committed:
build: bump jacoco to 0.8.11

cmark committed:
build: extract logback configuration file location to property

cmark committed:
chore(build): minor assembly description declaration change

cmark committed:
chore(tp): remove unnecessary protege xmlcatalog

cmark committed:
chore(tp): explicitly add jetty-session as tp dependency

cmark committed:
build: remove unused older jetty servlet and websocket api jars

cmark committed:
chore(deps): update elasticsearch8 client to 8.12.1

This updates parsson to 1.0.5 which resolves a security vulnerability
issue.

cmark committed:
chore(deps): upgrade netty to 4.1.106.Final

cmark committed:
chore(deps): upgrade kotlin to 1.9.22

cmark committed:
chore(deps): upgrade mapdb to 3.1.0

cmark committed:
chore(test): improve type consistenty in new query parameter encoder...

...helpers

apeteri commented on cis/com.b2international.snowowl.snomed.cis.product/assembly/common/bin/cis.sh:
```diff
@@ -80,6 +73,6 @@ exec $JAVA_EXECUTABLE $SO_JAVA_OPTS \
   -Dosgi.install.area="$KERNEL_HOME" \
   -Dosgi.configuration.area="$CONFIG_AREA" \
-  -jar plugins/org.eclipse.equinox.launcher_1.6.300.v20210813-1054.jar -console 2501
+  -jar plugins/org.eclipse.equinox.launcher_1.6.600.v20231106-1826.jar -console 2501
```
If `-Dosgi.console` works, it can be removed from this line (`cis.bat` is already updated this way):

```suggestion
  -jar plugins/org.eclipse.equinox.launcher_1.6.600.v20231106-1826.jar
```

apeteri commented on commons/com.b2international.collections.jackson/src/com/b2international/collections/deser/CollectionDeserializers.java:
```diff
@@ -188,7 +188,7 @@ public void deserializeContents(JsonParser jp, DeserializationContext ctxt,
                 } else {
                     if (t != JsonToken.VALUE_NULL) {
-                        throw ctxt.mappingException(_valueClass.getComponentType());
+                    	ctxt.reportWrongTokenException(_valueClass.getComponentType(), JsonToken.VALUE_NUMBER_INT, "Invalid token: expected int or float numbers");
```
I guess we can't exactly expect two things in this exception, but IMO `JsonToken.VALUE_NUMBER_FLOAT` would be a tiny bit more appropriate here as the main use case is deserialization of arrays containing floats.

apeteri commented on commons/com.b2international.index.test.tools/elasticsearch-default-memory-vm.options:
```diff
@@ -0,0 +1,3 @@
+-Xms2147483648
+-Xmx2147483648
```
Impressive numbers! 😄 Is there a special meaning attached to `Integer.MAX_VALUE`?

cmark replied:
I copied this from the `testcontainers:elasticsearch` module because it was unable to load this file from its jar during runtime in OSGi context. See the original file [here](https://github.com/testcontainers/testcontainers-java/blob/main/modules/elasticsearch/src/main/resources/elasticsearch-default-memory-vm.options)

apeteri replied:
It's in bytes, didn't notice this at first glance. So 2 GB.

cmark replied:
Yes :)

apeteri commented on core/com.b2international.snowowl.core.rest.tests/pom.xml:
```diff
@@ -59,9 +54,27 @@
 					<includes>
 						<include>**/AllSnowOwlApiTests.java</include>
 					</includes>
+					<skip>true</skip>
```
Is this test going to remain skipped?

cmark replied:
I think tests also fail due to the same Jetty 12 issue, but need to verify this on Linux machines as well. I can enable them temporarily and see if they pass in the GitHub Action env.

apeteri commented on core/com.b2international.snowowl.core.rest/.classpath:
```diff
@@ -1,29 +1,24 @@
+	<classpathentry exported="true" kind="lib" path="WEB-INF/lib/spring-security-config-6.2.1.jar"/>
+	<classpathentry exported="true" kind="lib" path="WEB-INF/lib/spring-aop-6.1.3.jar"/>
+	<classpathentry exported="true" kind="lib" path="WEB-INF/lib/spring-security-core-6.2.1.jar"/>
+	<classpathentry exported="true" kind="lib" path="WEB-INF/lib/spring-security-web-6.2.1.jar" sourcepath="C:/Users/mark/.m2/repository/org/springframework/security/spring-security-web/6.2.1/spring-security-web-6.2.1-sources.jar"/>
```
⚠️ Source attachment on classpath entry points to local path

cmark replied:
😭

apeteri commented on core/com.b2international.snowowl.core.rest/pom.xml:
```diff
@@ -103,8 +99,9 @@
 							</filesets>
-            			</configuration>
-          			</execution>
+							<skip>true</skip>
```
How about this cleaning step? Should it remain skipped?

cmark replied:
No, this was only to prevent failures in local builds running inside Eclipse. Thanks!

apeteri commented on core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/SnowOwlApiConfig.java:
```diff
@@ -214,18 +205,20 @@ protected void addFormatters(FormatterRegistry registry) {
 	
 	@Bean
-	public MethodValidationPostProcessor methodValidationPostProcessor() {
+	public static MethodValidationPostProcessor methodValidationPostProcessor() {
```
Does the introduction of this `static` modifier have any significance? Are you calling it outside Spring's DI?

cmark replied:
No, we do not call this outside Spring's DI. Spring while boots prints this warning message to the console if this method is not static.
```
[2024-02-09T13:20:14.299] WARN  main  s.c.s.PostProcessorRegistrationDelegate$BeanPostProcessorChecker Bean 'snowOwlApiConfig' of type [com.b2international.snowowl.core.rest.SnowOwlApiConfig$$SpringCGLIB$$0] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying). The currently created BeanPostProcessor [methodValidationPostProcessor] is declared through a non-static factory method on that class; consider declaring it as static instead. 
```

apeteri commented on core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/SnowOwlApiConfig.java:
```diff
@@ -214,18 +205,20 @@ protected void addFormatters(FormatterRegistry registry) {
-	    multipartResolver.setMaxUploadSize(httpConfig.getMaxRequestSizeBytes());
-	    multipartResolver.setMaxInMemorySize(httpConfig.getMaxInMemorySizeBytes());
-	    return multipartResolver;
+		// TODO add back max file size restrictions somehow
```
Looks like this is a Servlet-level configuration parameter now:

> Note: In order to use Servlet container based multipart parsing, you need to mark the affected servlet with a "multipart-config" section in web.xml, or with a `MultipartConfigElement` in programmatic servlet registration, or (in case of a custom servlet class) possibly with a `MultipartConfig` annotation on your servlet class. Configuration settings such as maximum sizes or storage locations need to be applied at that servlet registration level; a Servlet container does not allow for them to be set at the `MultipartResolver` level. 

* [Upgrading to Spring Framework 6.x](https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-6.x#web-applications-1)
* Javadoc of [StandardServletMultipartResolver](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/multipart/support/StandardServletMultipartResolver.html) (quote source)

apeteri replied:
I think this also means that the method can be removed entirely.

cmark replied:
Yes, thanks. I haven't got enough time to dig into this yet, but will do so now before merging the PR.

cmark replied:
Done

apeteri commented on snomed/com.b2international.snowowl.snomed.core.rest.tests/pom.xml:
```diff
@@ -72,9 +67,27 @@
 					<includes>
 						<include>**/AllSnomedApiTests.java</include>
 					</includes>
+					<skip>true</skip>
```
Also this one

apeteri commented on releng/com.b2international.snowowl.server.update/snow-owl-oss.product:
```diff
@@ -1,15 +1,15 @@
 <?pde version="3.5"?>
 
-<product name="Snow Owl Server" uid="com.b2international.snowowl.server" id="com.b2international.snowowl.server.product.product" application="com.b2international.snowowl.server.product.app" version="9.0.1.qualifier" useFeatures="true" includeLaunchers="true">
+<product name="Snow Owl Server" uid="com.b2international.snowowl.server" id="com.b2international.snowowl.server.product.product" application="com.b2international.snowowl.server.product.app" version="9.0.1.qualifier" type="features" includeLaunchers="true" autoIncludeRequirements="true">
```
Tycho does not seem to support `autoIncludeRequirements` yet (at least the issue is still open): https://github.com/eclipse-tycho/tycho/issues/1056

Do we need this option, is it helping with something within the IDE?

cmark replied:
I did not explicitly set this, probably it was the Eclipse editor updating the stuff behind the scenes. Not sure if we need it or not. The produced product looks good actually.

apeteri commented on fhir/com.b2international.snowowl.fhir.rest.tests/pom.xml:
```diff
@@ -88,6 +102,7 @@
 			            	<includes>
 								<include>**/AllFhirRestTests.java</include>
 							</includes>
+							<skip>true</skip>
```
And this one. 😸

cmark replied:
Yes, the three rest api tests failed on my computer during the maven build. I'll enable them now to see if they pass on our Unix build platform.

apeteri reviewed (changes requested):
Looks great at first glance! See feedback below.

cmark committed:
Update cis/com.b2international.snowowl.snomed.cis.product/assembly/common/bin/cis.sh

Co-authored-by: András Péteri <apeteri@b2international.com>

cmark committed:
fix(collections): report VALUE_NUMBER_FLOAT missing token...

...instead of incorrect VALUE_NUMBER_INT in
FloatCollectionDeserializerBase

cmark committed:
chore(dev): remove incorrect classpath source attachment path

cmark committed:
chore(dev): add m2e directive to ignore folder cleaning in core.rest

Due to project references Eclipse will report errors when m2e will try
to delete the unwanted but still referenced jars from the lib folder.

cmark committed:
build: enable rest api tests

nagyo commented on README.md:
```diff
@@ -82,21 +82,19 @@ These instructions will get Snow Owl up and running on your local machine for de
 
 Snow Owl is an Equinox-OSGi based server. To develop plug-ins for Snow Owl you need to use Eclipse as IDE: 
-* Download Eclipse IDE for Eclipse Committers 2021-12 package from here: https://www.eclipse.org/downloads/packages/release/2021-12/r/eclipse-ide-eclipse-committers
+* Download Eclipse IDE for Eclipse Committers 2023-09 package from here: https://www.eclipse.org/downloads/packages/release/2023-09/r/eclipse-ide-eclipse-committers
```
This could already be 2023-12

nagyo commented on README.md:
```diff
@@ -82,21 +82,19 @@ These instructions will get Snow Owl up and running on your local machine for de
 
 *Note: you may have to untick the `Show only the latest versions of the available software` checkbox to get older versions of a feature. Please use the exact version specified below, not the latest point release.*
 
+* Groovy Development Tools (https://groovy.jfrog.io/artifactory/plugins-release/e4.29)
```
With the eclipse version upgrade, you can use the e4.30 update site

nagyo commented on README.md:
```diff
@@ -82,21 +82,19 @@ These instructions will get Snow Owl up and running on your local machine for de
+  * Eclipse Groovy Development Tools - 5.1.0 (in category "Main Package")
+* MWE2 (https://download.eclipse.org/modeling/emft/mwe/updates/releases/2.12.1/)
+  * MWE SDK 1.6.1 (MWE)
 * Xtext/Xtend (https://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.25.0/)
```
Also, this one is 2.33.0, isn't it?

cmark replied:
Right, this should be 2.33.0.

nagyo commented on cis/com.b2international.snowowl.snomed.cis.product/assembly/common/bin/cis.sh:
```diff
@@ -80,6 +73,6 @@ exec $JAVA_EXECUTABLE $SO_JAVA_OPTS \
   -Dosgi.install.area="$KERNEL_HOME" \
   -Dosgi.configuration.area="$CONFIG_AREA" \
-  -jar plugins/org.eclipse.equinox.launcher_1.6.300.v20210813-1054.jar -console 2501
+  -jar plugins/org.eclipse.equinox.launcher_1.6.600.v20231106-1826.jar
```
Does the removal of the -console parameter come from the fact that the default is the same? We still need the osgi console capability in both the CIS and the OSS product.

nagyo replied:
I see that we are not using that anymore in the OSS product, I guess it is a historical residue 👍🏻

cmark replied:
Yes

nagyo commented on core/com.b2international.snowowl.core.rest.tests/src/configuration/jetty-deploy.xml:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
```
I thought we wouldn't need to copy these files, but I might be outdated with the jetty and osgi boot issue?

cmark replied:
There is another issue with osgi.boot on Jetty 12. The only workaround now is to use explicit jetty files in all our e2e tests.

nagyo reviewed (approved):
LGTM! (as much as I was able to go through all the files 😅)

nagyo committed:
chore(releng): remove JDK and Maven version from Jenkins pipeline

- add job parameters to Jenkinsfile

cmark committed:
chore(deps): upgrade ecl to 2.1.5

cmark committed:
chore(deps): upgrade owlapi to 4.5.26.b2i

cmark committed:
chore(deps): upgrade protege to 5.0.7.b2i

cmark committed:
chore(deps): use released version of owltoolkit 5.1.1

cmark committed:
chore(docs): update Eclipse IDE version

Update Groovy Development Tools update site.
Update MWE versions and update site.
Update Xtext versions and update site.

apeteri commented on README.md:
```diff
@@ -82,21 +82,19 @@ These instructions will get Snow Owl up and running on your local machine for de
 
 Snow Owl is an Equinox-OSGi based server. To develop plug-ins for Snow Owl you need to use Eclipse as IDE: 
-* Download Eclipse IDE for Eclipse Committers 2021-12 package from here: https://www.eclipse.org/downloads/packages/release/2021-12/r/eclipse-ide-eclipse-committers
+* Download Eclipse IDE for Eclipse Committers 2023-09 package from here: https://www.eclipse.org/downloads/packages/release/2023-12/r/eclipse-ide-eclipse-committers
```
Now it's 2023-09 and 2023-12 at the same time.

cmark replied:
Fixed

apeteri reviewed (approved):
OK to merge once the README is adjusted and (possibly) the multipart upload configuration is added back to its new place.

cmark committed:
fix(api): add back http multipart configuration based on snowowl.yml

Location will no longer be set by default to `/tmp`, let the servlet
container (Jetty) default temporary directory be used instead of a
custom directory. The multipart element is still needed in the web.xml
to kick-off the necessary servlet initialization logic. Actual max
file/request size and file threshold limits will be set when Spring
initializes the rest api webapp.

cmark committed:
chore(docs): update latest tested Eclipse version
